File: app/src/main/java/com/example/habitpalette/CurrentHabitRecyclerAdapter.java
package com.example.habitpalette;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class CurrentHabitRecyclerAdapter extends RecyclerView.Adapter<CurrentHabitRecyclerAdapter.ViewHolder> {

    private ArrayList<CurrentHabitRecyclerItem> mData = null;

    CurrentHabitRecyclerAdapter(ArrayList<CurrentHabitRecyclerItem> list) {
        mData = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View view = inflater.inflate(R.layout.item_current_habit, parent, false);
        CurrentHabitRecyclerAdapter.ViewHolder vh = new CurrentHabitRecyclerAdapter.ViewHolder(view);

        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        CurrentHabitRecyclerItem item = mData.get(position);
        String period= Integer.toString(item.getHabitPeriod());
        holder.habitPeriod.setText("습관 "+period+"일째");
        holder.habitTitle.setText(item.getHabitTitle());
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView habitTitle;
        TextView habitPeriod;

        ViewHolder(View itemView) {
            super(itemView);

            habitTitle = itemView.findViewById(R.id.text_habit_title);
            habitPeriod = itemView.findViewById(R.id.text_habit_period);
        }
    }
}

File: app/src/main/java/com/example/habitpalette/ui/habit/CurrentHabitRecyclerAdapter.kt
package com.example.habitpalette.ui.habit

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.example.habitpalette.data.model.Habit
import com.example.habitpalette.databinding.ItemCurrentHabitBinding

class CurrentHabitRecyclerAdapter(
    var currentList: List<Habit>,
    val onClickDetailButton: () -> Unit) :
    RecyclerView.Adapter<CurrentHabitRecyclerAdapter.CurrentHabitViewHolder>() {

    private lateinit var itemBinding: ItemCurrentHabitBinding

    inner class CurrentHabitViewHolder(private val itemBinding: ItemCurrentHabitBinding): RecyclerView.ViewHolder(itemBinding.root){
        fun bind(data: Habit) {
            itemBinding.tvTitle.text = data.title
            itemBinding.tvPeriod.text = data.duration.toString()
            itemBinding.pvCurrentScore.percent = 50f

            // item 클릭해서 CalendarActivity로 intent
            itemBinding.layoutCurrentItem.setOnClickListener {
                onClickDetailButton.invoke()
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CurrentHabitViewHolder {
        itemBinding = ItemCurrentHabitBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CurrentHabitViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: CurrentHabitViewHolder, position: Int) {
        holder.bind(currentList[position])
    }

    override fun getItemCount(): Int {
        return currentList.size
    }
}